pydantic/_internal/_generate_schema.py

```diff
@@ -318,11 +356,24 @@ def generate_td_field_schema(
         """
         assert field_info.annotation is not None, 'field_info.annotation should not be None when generating a schema'
         schema = self.generate_schema(field_info.annotation)
+        this_field_validators = filter_field_decorator_info_by_field(decorators.validator.values(), name)
```

Comment:
It looks like `this_field_validators` is part of the v1 compatibility shim; if that's right, can we move this line below the comment below?